themes/_core/js/product.js

```diff
@@ -191,23 +191,43 @@ function updateProduct(event, eventType, updateUrl) {
-          $(prestashop.selectors.product.discounts).first().replaceWith(data.product_discounts);
-          $(prestashop.selectors.product.additionalInfos).first().replaceWith(data.product_additional_info);
-          $(prestashop.selectors.product.details).replaceWith(data.product_details);
-          $(prestashop.selectors.product.flags).first().replaceWith(data.product_flags);
-          replaceAddToCartSections(data);
-          const minimalProductQuantity = parseInt(data.product_minimal_quantity, 10);
-
-          document.dispatchEvent(updateRatingEvent);
+        if ($('.quickview .images-container, .page-product:not(.modal-open) .row .images-container, .page-product:not(.modal-open) .product-container .images-container').html() !== $newImagesContainer.find('.quickview .images-container, .page-product:not(.modal-open) .row .images-container, .page-product:not(.modal-open) .product-container .images-container').html()) {
+          $('.quickview .images-container, .page-product:not(.modal-open) .row .images-container, .page-product:not(.modal-open) .product-container .images-container').replaceWith(data.product_cover_thumbnails);
+        }
+
+        $(
+          '.quickview .product-prices, .page-product:not(.modal-open) .row .product-prices, .page-product:not(.modal-open) .product-container .product-prices',
+        )
+          .first()
+          .replaceWith(data.product_prices);
+        $(
+          '.quickview .product-customization, .page-product:not(.modal-open) .row .product-customization, .page-product:not(.modal-open) .product-container .product-customization',
+        )
+          .first()
+          .replaceWith(data.product_customization);
+        $(
+          '.quickview .product-variants, .page-product:not(.modal-open) .row .product-variants, .page-product:not(.modal-open) .product-container .product-variants',
+        )
+          .first()
+          .replaceWith(data.product_variants);
+        $(
+          '.quickview .product-discounts, .page-product:not(.modal-open) .row .product-discounts, .page-product:not(.modal-open) .product-container .product-discounts',
+        )
+          .first()
+          .replaceWith(data.product_discounts);
+        $(
+          '.quickview .product-additional-info, .page-product:not(.modal-open) .row .product-additional-info, .page-product:not(.modal-open) .product-container .product-additional-info',
+        )
+          .first()
+          .replaceWith(data.product_additional_info);
+        $('.quickview #product-details, #product-details').replaceWith(data.product_details);
+        $(
+          '.quickview .product-flags, .page-product:not(.modal-open) .row .product-flags, .page-product:not(.modal-open) .product-container .product-flags',
+        )
+          .first()
+          .replaceWith(data.product_flags);
+        replaceAddToCartSections(data);
+        const minimalProductQuantity = parseInt(data.product_minimal_quantity, 10);
```

Comment:
You choosed the wrong one, you should keep selectors map in place

Reply:
It was fifty/fifty 😭

Reply:
Fixed, I believe, in https://github.com/PrestaShop/PrestaShop/pull/20624/commits/ae4811076c8e407db5100ec89a08502a2f2648df